spinalcordtoolbox/image.py

```diff
@@ -55,6 +55,49 @@ def _get_permutations(im_src_orientation, im_dst_orientation):
     return perm, inversion
 
 
+def rpi_slice_to_orig_orientation(dim, orig_orientation, slice_number, axis):
+    """
+    Convert slice number from RPI (right-posterior-inferior) to original orientation, e.g., AIL
+    (anterior-inferior-left).
+    :param dim: tuple, dimensions of the image in RPI orientation.
+    :param orig_orientation: str, original image e.g., AIL.
+    :param slice_number: str, slice number in RPI orientation, e.g., 9.
+    :param axis: str, axis of the slice in the RPI orientation. 'x' corresponds to the R-L axis,
+    'y' to the A-P axis, and 'z' to the I-S axis.
+    :return: int, slice number in original orientation, e.g., 6.
+
+    Example: considering an image with 20 sagittal slices (0-19) and sagittal slice number 13 in the AIL orientation,
+    the corresponding slice number in the RPI orientation is 6:
+            rpi_slice_to_orig_orientation((20, 640, 640), 'AIL', 13, 'x') -> 6
+    Note: we use 'x' in this example as it corresponds to the R-L direction (first in RPI --> 'x')
+    """
+    # Ensure slice_number is an integer
+    slice_number = int(slice_number)
+
+    # Get dimensions of the image in RPI orientation
+    nx, ny, nz = dim
+
+    # Define axis index mapping
+    axis_map = {'x': 0, 'y': 1, 'z': 2}
+    axis_index = axis_map[axis]
+
+    # Get the permutations and inversions
+    _, inversion = _get_permutations('RPI', orig_orientation)
+
+    # Calculate the corresponding slice number in the original orientation
+    if inversion[axis_index] == 1:
+        orig_slice_number = slice_number
+    else:
+        if axis_index == 0:
+            orig_slice_number = nx - 1 - slice_number
+        elif axis_index == 1:
+            orig_slice_number = ny - 1 - slice_number
+        elif axis_index == 2:
+            orig_slice_number = nz - 1 - slice_number
+
+    return orig_slice_number
```

Comment:
I think you might be able to make this much shorter if you accept axis numbering (like numpy does; `axis={0,1,2}`)?

```suggestion
    _, inversion = _get_permutations('RPI', orig_orientation)
    return (dim[axis] - 1 - slice_number) if inversion[axis] == -1 else slice_number
```

Reply:
hmm, interesting! Will the simplified code work for other orientations than `AIL`? I'll test this tmr!

Reply:
It should! Basically, instead of passing `'x'`, `'y'`, etc. you just pass `0`/`1`/`2`, so the callers would need to be changed too.

Reply:
Okay, the simplification works! Thank you, Joshua! Implemented in https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4597/commits/6ee8d5a77dfa1649a822811629448876e3ede4e6. 
However, it seems to be working only for `AIL` :-(

When trying on `SAR`, it does not renumber the slices correctly...

Commands I used to create testing `SAR` images: 

```console
cd sci-zurich/derivatives/labels/sub-zh102/ses-01/anat/
mkdir test_sar
cp sub-zh102_ses-01_acq-sag_T2w_seg-manual.nii.gz sub-zh102_ses-01_acq-sag_T2w_lesion-manual.nii.gz test_sar
cd test_sar
sct_image -i sub-zh102_ses-01_acq-sag_T2w_lesion-manual.nii.gz -setorient SAR -o sub-zh102_ses-01_acq-sag_T2w_lesion-manual.nii.gz
sct_image -i sub-zh102_ses-01_acq-sag_T2w_seg-manual.nii.gz -setorient SAR -o sub-zh102_ses-01_acq-sag_T2w_seg-manual.nii.gz
```

Reply:
I guess we need to consider also the permutations; I tried the following code but it does not work properly.

```python
# Get the permutations and inversions
perm, inversion = _get_permutations('RPI', self.orientation)

# Get the corresponding permuted axis
permuted_axis_index = perm[axis]

# Convert slice number based on permutation and inversion
return (dim[axis] - 1 - slice_number) if inversion[permuted_axis_index] == -1 else slice_number
```

Reply:
I think there might be an even simpler approach? Since we know the input is RPI, we only have to invert if `LAS` is in the original orientation:

```Python
flipped_axis = 'LAS'
return (dim[axis] - 1 - slice_number) if flipped_axis[axis] in orig_orientation else slice_number
```

We don't need to use `_get_permutations` at all?

Reply:
Yeah, you are right again! 💯  Done in https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4597/commits/4bfea3e5df8bcecefb066c4c95ea91cfaf184e81

Reply:
sorry, I got completely confused....

`if flipped_axis[axis] in orig_orientation` will be actually always `True` because `flipped_axis` == `orig_orientation `

Reply:
`flipped_axis` != `orig_orientation`? I thought "flipped_axis" was the flipped version of RPI (i.e. LAS), whereas orig_orientation may be SAR, AIL, etc.

Reply:
Okay, I pushed the code which is working for both `AIL` and `SAR` in https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4597/commits/dc7bda9169532d4124515c59f9965f52118490cf